tests/xep206.py

```diff
@@ -47,3 +73,59 @@ def _testSessionCreate(res):
         d = self.proxy.connect(BOSH_XML).addCallback(_testSessionCreate)
         return d
 
+
+class DeliveryFailureTestCase(XEP0206TestCase):
+    """
+    Tests for the XEP0206 message delivery failure NACK feature.
+    """
+
+    def setUp(self):
+        self.xmlstream = DummyXmlStream()
+        self.session = session.Session(
+            DummyPint(),
+            {
+                'rid': 0,
+                'to': '',
+                'route': 'xmpp:127.0.0.1:8080',
+            },  # Dummy values to satisfy the constructor. Not needed in tests.
+        )
+        self.session.xmlstream = self.xmlstream
+
+    def tearDown(self):
+        del self.session
+
+    def test_skips_presence(self):
+        """
+        Test that the NACK skips presence elements.
+        """
+        sentinel = domish.Element((None, 'presence'))
+        self.session.elems.append(sentinel)
+        self.session.disconnect()
+        assert sentinel not in self.xmlstream.elements
```

Comment:
Ah yeah. I was stuck in pytest mode when I wrote this. The `assertNotIn` is PY27 and above. Are you still trying to maintain PY26 compat? If so I can either convert this to `assertTrue` or add a customer assertion error string. Something along the lines of:

```python
assert sentinel not in self.xmlstream.elements, "{} found in {}".format(
    sentinel,
    self.xmlstream.elements
)
```